lib/spack/spack/cmd/clean.py

```diff
@@ -88,3 +93,17 @@ def clean(parser, args):
     if args.misc_cache:
         tty.msg('Removing cached information on repositories')
         spack.caches.misc_cache.destroy()
+
+    if args.python_cache:
+        tty.msg('Removing python cache files.')
+        for root, dirs, files in os.walk(os.environ['SPACK_ROOT']):
```

Comment:
This environment variable is not set by default, only if you happen to use `setup-env.sh`. You should use this instead:
```python
from spack.paths import spack_root

for root, dirs, files in os.walk(spack_root):
```